README.md

```diff
@@ -32,21 +32,101 @@ Before you run the tests, you'll need a full Deis cluster up and running in Kube
 
 ## Run the Tests
 
-To run the entire test suite:
+There are three options for how to execute the tests.
+
+### Remotely
+
+If you have Go 1.5 or greater already installed and working properly and also have the [Glide](https://github.com/Masterminds/glide) dependency management tool for Go installed, you may clone this repository into your `$GOPATH`:
+
+```console
+git clone git@github.com:deis/workflow-e2e.git $GOPATH/src/github.com/deis/workflow-e2e
+```
+
+One-time execution of the following will resolve the test suite's own dependencies:
+
+```console
+$ make bootstrap
+```
+
+Export a value for the `DEIS_CONTROLLER_URL` environment variable that matches the controller endpoint you would normally use with the `deis register` or `deis login` commands:
+
+```console
+$ export DEIS_CONTROLLER_URL=http://deis.your.cluster
+```
+
+To execute the entire test suite:
 
 ```console
 $ make test-integration
 ```
 
-To run a single test or set of tests, you'll need the [ginkgo](https://github.com/onsi/ginkgo) tool installed. You can then use the `--focus` option:
+Tests execute in parallel by default.  If you wish to control the number of executors, export a value for the `GINKGO_NODES` environment variable:
+
+```console
+$ export GINKGO_NODES=15
+```
+
+If this is not set, Ginkgo will automatically choose a number of executors based on the number of CPU cores _on the machine from which you execute the tests_.  It is important to note that the cluster you are testing against may be able to sustain either fewer or more than that number.
+
+Setting the `GINKGO_NODES` environment variable to a value of `1` will allow serialized execution of all tests in the suite.
+
+To run a single test or set of tests, you'll need the [ginkgo](https://github.com/onsi/ginkgo) tool installed on your machine:
+
+```console
+$ go get github.com/onsi/ginkgo/ginkgo
+```
+
+You can then use the `--focus` option to run subsets of the test suite:
+
+```console
+$ ginkgo --focus="deis apps" tests
+```
+
+### Remotely, In a Container
+
+If you do not have Go 1.5 or greater installed locally, but do have a Docker daemon running locally (or are using docker-machine), you can quite easily execute tests against a remote cluster from within a container.
+
+First, export a value for the `DEIS_CONTROLLER_URL` environment variable that matches the controller endpoint you would normally use with the `deis register` or `deis login` commands:
+
+```console
+$ export DEIS_CONTROLLER_URL=http://deis.your.cluster
+```
+
+Then build the test image and execute the test suite:
+
+```console
+$ make docker-build docker-test-integration
+```
+
+Tests execute in parallel by default.  If you wish to control the number of executors, export a value for the `GINKGO_NODES` environment variable:
```

Comment:
nit: it appears the same handful of paragraphs on `GINKGO_NODES`/parallel vs. serial appear in two places -- maybe just move it to its own section of the README? (If/when the description needs updating, we'll only have to do it in one place)